src/screens/ProfileScreen/elements/BioAndDMSetting.tsx

```diff
@@ -76,68 +108,105 @@ const BioAndDMSetting: React.FC<BioAndDMSettingProps> = ({
+              style={{
+                display: 'flex',
+                flexDirection: 'row',
+                justifyContent: 'center',
+                alignItems: 'center'
+              }}>
+              <PencilIcon />
+              <Text
+                style={{
+                  fontSize: 14,
+                  fontWeight: '500',
+                  color: COLORS.white,
+                  marginLeft: 4
+                }}>
+                Edit
+              </Text>
+            </View>
+          </Pressable>
         </View>
-      </Pressable>
-
-      <TouchableOpacity onPress={toggleSwitchAnon} style={styles.toggleSwitchAnon}>
-        <ToggleSwitch
-          value={isAnonymity}
-          onValueChange={toggleSwitchAnon}
-          labelLeft="Allow anonymous messages?"
-          circleActiveColor={COLORS.signed_primary}
-          activeTextColor={COLORS.signed_primary}
+        <Divider
+          style={{
+            marginVertical: 12
+          }}
         />
-      </TouchableOpacity>
-
-      {isAnonymity && (
-        <TouchableOpacity
-          onPress={toggleSwitchAnonAllowFollowing}
-          style={styles.toggleSwitchAnonFollowing}>
-          <ToggleSwitch
-            value={isAllowFollowingSendDM}
-            onValueChange={toggleSwitchAnonAllowFollowing}
-            labelLeft="Only allow anon DMs from users you follow?"
-            circleActiveColor={COLORS.signed_primary}
-            activeTextColor={COLORS.signed_primary}
-          />
+        <Text style={styles.bioText}>{addDotAndRemoveNewline(bio)}</Text>
+      </View>
+      <View
+        style={{
+          backgroundColor: COLORS.white,
+          borderRadius: 12,
+          padding: 12,
+          marginHorizontal: 12,
+          shadowColor: 'rgba(0, 0, 0, 0.04)',
+          shadowOffset: {
+            width: 0,
+            height: 1
+          },
+          shadowOpacity: 1,
+          shadowRadius: 2,
+          elevation: 2,
+          marginTop: -50,
+          borderWidth: 1,
+          borderColor: COLORS.gray
+        }}>
+        <Text
+          style={{
+            fontSize: 12,
+            fontWeight: '400',
+            color: COLORS.gray500,
+            marginBottom: 12
+          }}>
+          Other users will be able to reply to your prompt and direct message you.
+        </Text>
+        <TouchableOpacity onPress={toggleSwitchAnon}>
+          <CheckBoxCustom value={isAnonymity} label="Allow anonymous messages?" />
         </TouchableOpacity>
-      )}
+
+        {isAnonymity && (
+          <>
+            <Divider style={{marginVertical: 12, backgroundColor: COLORS.gray}} />
+            <TouchableOpacity onPress={toggleSwitchAnonAllowFollowing}>
+              <CheckBoxCustom
+                value={isAllowFollowingSendDM}
+                label="Only allow anon DMs from users you follow?"
+              />
+            </TouchableOpacity>
+          </>
+        )}
+      </View>
     </View>
   );
 };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made to the render method of `BioAndDMSetting` are significant. The new implementation seems to be more user-friendly and intuitive. However, there are some magic numbers used in the styling (e.g., line 139, 146, 149, etc.). Consider defining these values as constants at the top of your file or in a separate constants file. This will make your code easier to maintain and understand.



<!-- This is an auto-generated comment by OSS CodeRabbit -->